src/hooks/useTransactionHistory.ts

```
@@ -222,8 +227,10 @@ export function useTransactionHistory({
 
     // Latest transactions mode (for home page)
     if (mode === 'latest') {
+        // if filterMutualTxs is true, we need to add the username to the query key to invalidate the query when the username changes
+        const queryKeyTxn = TRANSACTIONS + (filterMutualTxs ? username : '')
         return useQuery({
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)